### PR TITLE
Add integration tests for pass_through_controllers

### PR DESCRIPTION
**Pull request by stefanscherzinger**
This PR does the following:
- It refactors the previous _example_ folder into a _test_ folder and removes
everything that becomes obsolete through using _cartesian_trajectory_controller_'s dummy setup
- It defines a simple integration test (normal execution) for both
joint-based and Cartesian trajectory forwarding.

### CODE REVIEW

File: cartesian_trajectory_controller/CMakeLists.txt [hunk 1]
```diff
@@ -203,6 +203,14 @@ install(FILES
   ${PROJECT_NAME}_plugin.xml
   DESTINATION ${CATKIN_PACKAGE_SHARE_DESTINATION}
 )
+# Install test setup files for packages that re-use them
```
Comment (fmauch) [inline cartesian_trajectory_controller/CMakeLists.txt]: Wouldn't it make more sense to define a separate test package that can be used by different packages?

**Review by fmauch — COMMENTED** [review]
Wouldn't it make more sense to define a separate test package that can be used by different packages?

I think, it also would be a lot cleaner to split this into two separate PRs, as this is getting quite large with a lot of changes not being directly connected to the passthrough controllers.

**Comment by stefanscherzinger** [comment]
> Wouldn't it make more sense to define a separate test package that can be used by different packages?

I get your idea but that would introduce yet another package. It's mostly four files that are being reused. I'm slightly against putting them into a separate package..

> would be a lot cleaner to split this into two separate PRs, as this is getting quite large with a lot of changes not being directly connected to the passthrough controllers.

That's true. I'll try to change this PR to affect the pass_through_controllers subpackage only.

**Comment by stefanscherzinger** [comment]
@fmauch [Here's a separate PR](https://github.com/fzi-forschungszentrum-informatik/cartesian_ros_control/pull/23) for the changes to the cartesian_trajectory_controller. I'll remove those changes from here once that got merged.

File: cartesian_trajectory_controller/CMakeLists.txt (hunk 1 above)
Comment (stefanscherzinger) [reply cartesian_trajectory_controller/CMakeLists.txt]: ... I'm hesitating. It's really not much files and they fit well into the cartesian_trajectory_controller package.

**Review by fmauch — APPROVED** [review]
Looks good to me.

**Pull request merged**